### PR TITLE
Animated wheels

### PR DESCRIPTION
**Pull request by AnshJuneja**

### CODE REVIEW

**Comment by AnshJuneja** [comment]
diff1

diff2


These 2 pictures show the diff between the CustomCarPawn blueprint file (old on right half of the picture and new on left half). As you can see, the wheel meshes and the rotation movement variables are added in the new blueprint.

File: Unreal/Plugins/AirSim/Source/Vehicles/CustomCar/CustomCarPawn.cpp [hunk 1]
```diff
@@ -205,6 +222,25 @@ void ACustomCarPawn::Tick(float Delta)
...
+    else { // Set tire speed
+        rotating_movement_fl_->RotationRate.Roll = vehicle_state_.fl_wheel_state.angular_velocity*RAD2DEG;
+        rotating_movement_fr_->RotationRate.Roll = vehicle_state_.fr_wheel_state.angular_velocity*RAD2DEG;
+        rotating_movement_rl_->RotationRate.Roll = vehicle_state_.rl_wheel_state.angular_velocity*RAD2DEG;
+        rotating_movement_rr_->RotationRate.Roll = vehicle_state_.rr_wheel_state.angular_velocity*RAD2DEG;
+
+        rotating_movement_fl_->SetUpdatedComponent(wheel_fl_);
+        rotating_movement_fr_->SetUpdatedComponent(wheel_fr_);
+        rotating_movement_rl_->SetUpdatedComponent(wheel_rl_);
+        rotating_movement_rr_->SetUpdatedComponent(wheel_rr_);
```
Comment (mantkiew) [inline Unreal/Plugins/AirSim/Source/Vehicles/CustomCar/CustomCarPawn.cpp:240]: so, the docs say that the movement component automatically registers itself with a tick and gets executed automatically. Can we update the component with the new speed outside of the tick function? That would solve the issue.

File: Unreal/Plugins/AirSim/Source/Vehicles/CustomCar/CustomCarPawn.cpp [hunk 2]
```diff
@@ -84,6 +86,11 @@ void ACustomCarPawn::setupVehicleMovementComponent()
 void ACustomCarPawn::setVehicleModelInput(VehicleInput vehicle_input)
 {
     vehicle_model_.setVehicleInput(vehicle_input);
+    float steering = vehicle_input.steering_angle; 
+    if (steering < -8.48f) steering = -8.48f;
+    else if (steering > 8.48f) steering = 8.48f;
```
Comment (mantkiew) [inline Unreal/Plugins/AirSim/Source/Vehicles/CustomCar/CustomCarPawn.cpp]: there is a "clamp" function that can be used for this.

**Review by mantkiew — CHANGES_REQUESTED** [review]

File: Unreal/Plugins/AirSim/Source/Vehicles/CustomCar/CustomCarPawn.cpp (hunk 1 above)
Comment (AnshJuneja) [reply Unreal/Plugins/AirSim/Source/Vehicles/CustomCar/CustomCarPawn.cpp:240]: This still causes it to run in the same tick as the tire yaw setting code, I tried and it doesn't work :/

File: Unreal/Plugins/AirSim/Source/Vehicles/CustomCar/CustomCarPawn.cpp (hunk 2 above)
Comment (AnshJuneja) [reply Unreal/Plugins/AirSim/Source/Vehicles/CustomCar/CustomCarPawn.cpp]: Added this

**Review by mantkiew — APPROVED** [review]
Works.

**Review by mantkiew — CHANGES_REQUESTED** [review]
Why does the front-right wheel look weird? 

right-side wheels

left-side wheels

**Review by mantkiew — APPROVED** [review]

**Pull request merged**